arch/sim/src/sim/up_head.c

```diff
@@ -77,6 +78,8 @@ static char g_logbuffer[4096];
 
 int main(int argc, char **argv, char **envp)
 {
+  process_cli_options(argc, argv);
```

Comment:
How about we just save argc and argv in some global variable? Then each component could invoke getopt to parse the argument who is interesting. The advantage is that:
1.Don't need expose the component internal detail to the world
2.Don't need modify the common code to add the new option

Reply:
@xiaoxiang781216 with your suggestion, how can component-A skip options for component-B?

Reply:
Just ignore the option he doesn't understand.

Reply:
getopt doesn't work that way. or are you suggesting to stop using getopt?

Reply:
> getopt doesn't work that way. or are you suggesting to stop using getopt?

Why we can't use getopt like this:
```
  while ((option = getopt(argc, argv, "f")) != ERROR)
    {
      if (option == 'f')
        {
           ...
        }
   }
```

Reply:
@xiaoxiang781216 
how would it skip an unknown option, say, -g?
note that -g might or might not take an argument.

Reply:
> @xiaoxiang781216
> how would it skip an unknown option, say, -g?
> note that -g might or might not take an argument.

Yes, getopt can't handle the option with argument. But I still think the code struct is better even we have to parse the argument directly, do you think so?

Reply:
it depends.
if we invent a "generic" framework, how many components would use it?
if it's just a few or several, i guess #ifdef is good enough.

Reply:
BTW, is it enough that we just provide Kconfig like tap bridge?

Reply:
maybe. a command line option is more convenient for me though.

may i move the command line option part to a separate PR? i haven't expected it was this controversial.

Reply:
> maybe. a command line option is more convenient for me though.
> 
> may i move the command line option part to a separate PR? i haven't expected it was this controversial.

Sure, thanks for your support.
BTW, how about we use environment variable which is more easier than command line for extension.

Reply:
i agree environment variables don't have modularity issues. i feel it's a bit backdoor-ish though.